core/src/muxing.rs

```diff
@@ -83,6 +83,9 @@ pub trait StreamMuxer {
     /// Future that will be resolved when the outgoing substream is open.
     type OutboundSubstream;
 
+    /// Error type of the muxer
+    type Error;
```

Comment:
By curiosity, did you try adding a `Into<io::Error>` bound here? This might make it possible to remove lots of requirements from elsewhere.

Reply:
actually didn't think of it, thanks